base/compiler/abstractinterpretation.jl

```diff
@@ -1450,6 +1508,27 @@ function typeinf_local(interp::AbstractInterpreter, frame::InferenceState)
     nothing
 end
 
+function conditional_changes(changes::VarTable, @nospecialize(typ), var::Slot)
+    if typ ⊑ (changes[slot_id(var)]::VarState).typ
```

Comment:
The original code was https://github.com/JuliaLang/julia/blob/d46198d1bbc19165e45f0b0a7003253ccff0773b/base/compiler/abstractinterpretation.jl#L1322-L1325

I changed that so that we propagate new conditional type only when it is really worthwhile by checking it's lower in the lattice than the original type.

Reply:
I mostly assumed that was assured by construction before (since we build these with typesubtract and typeintersect, or explicitly with Any), but makes sense also to be explicit